### ISSUE TITLE
Skipping intros of shows

### ISSUE THREAD

**Issue by okan35** [nova-video-player/aos-AVP#610]
It would be amazing to have this feature. Basically like Netflix but could be an option to check so it would either pass like sponsorblock or just show a button to pass intro like on Netflix.

I am not sure which way is the easiest to do this since nova does not have any backend support.

But maybe this could be used if the video file has lets say intro chapter and nova would detect this and automatically pass it. To me this seems the easiest way to do it for nova.

**Comment by Pentaphon** [comment]
Every intro is different so this is something that an open source player doesn't have the resources to make work for every video. A sponsorblock thing would never happen since that would imply piracy. I think we are just going to have to keep skipping forward as we normally do. There's no player in the world that can do this other than a generic "skip the first minute of every video" command.

**Comment by mschumacher69** [comment]
> Every intro is different so this is something that an open source player doesn't have the resources to make work for every video. A sponsorblock thing would never happen since that would imply piracy. I think we are just going to have to keep skipping forward as we normally do. There's no player in the world that can do this other than a generic "skip the first minute of every video" command.

Agreed, scene releases don't have chapters, so I don't believe there's a way to do this.

**Comment by okan35** [comment]
@courville 

How about having a txt file for each episode like subtitles but the info would be about timestamps that should be skipped. I believe since there is no backend for nova and reencoding a file to remove intro is too much work, this seems like the most viable option.

Of course the work would be done by a script which would output necessary files that have timestamps to skip.

And I would personally work on such script if you also agree as well. Just that you would need new columns in DBs for timestamps to pass and also applying that to player would be some work.

**Comment by courville** [comment]
@okan35, I am all for this feature of course. I thought about it already and IMHO there is no need to add a column in the mediadb and we can indeed process an txt/xml file providing timestamps of the intro and outro and have nova skip these as an option.
As you pointed out, it would rely on an external process to construct this txt/xml file (like the xml bookmarks).
For the clever script, I found these references that could be of interest:
* https://github.com/Hellowlol/bw_plex
* https://github.com/xbmc/notrobro
* https://docs.google.com/document/d/1AXIyxK6p3m_IQBWOgBDXxaUFi_457pSblPUQti9uEpo/edit
* https://github.com/nielstenboom/recurring-content-detector
If you can find some time to work on this script, I would be happy to make it happen on nova.

**Comment by okan35** [comment]
I have checked some of those projects and hopefully instead of rewriting one, I will adapt one of them to produce timestamps where an intro starts and ends.

**Comment by okan35** [comment]
I was able to create some timestamps I believe from the project notrobro but I dont get what kind of timestamps are these. It gives outputs with .edl files which has some strange time format if even they are for timestamps. 

Do you have any knowledge about edl files ?

Because if this works I could create an executable with all the dependencies and you could try.

**Comment by courville** [comment]
@okan35 could you please post an edl file you generated? I will have a look.

**Comment by courville** [comment]
FTR another project https://github.com/JacobEkedahl/detect-intros-from-video
Paper related https://www.diva-portal.org/smash/get/diva2:1412560/FULLTEXT01.pdf

**Comment by okan35** [comment]
I will send you edl files in the evening but from my research about it, it seems the timestamps I got are not correct since that notrobro does not use audio fingerprint but it uses frame comparison. 

So I guess it is not that accurate, will need to find a project that is accurate or learn how to do it from zero :)

**Comment by okan35** [comment]
github does not let me upload here but it is basically like this.

s01e01.mkv and s01e01.edl with data below
210.335 242.909 4
1339.003 1339.003 5

s01e02.mkv and s01e02.edl with data below
179.638 232.023 4
1276.876 1276.876 5

s01e03.mkv and s01e03.edl with data below
112.946 122.914 4
1252.615 1277.2640000000001 5 


From what I checked these are timestamps in seconds and first lines are for intro, second lines are for outro.

**Comment by courville** [comment]
OK, I tried to see if adding this skipping strategy could be a quick win on nova side, turns out that it can infer some quite invasive recoding to be done. It will take some time (i.e. I will not address it right away).

**Comment by okan35** [comment]
yeah, no worries, I am sure it would be necessary to do some recoding.

 I think easiest way to implement this would be the way resume works which is basically reading the xml for episode. 
So when an episode is started app would also need to check and parse the intro timestamps and just skip those parts if not skipped already.

**Comment by Pentaphon** [comment]
How would this work if not with some kind of Sponsorblock-type server that could be construed as implicitly encouraging piracy?
